### DISCUSSION TITLE
[@types/react-table] `Accessor` function type requires non-existant property in `sub` parameter

### DISCUSSION THREAD

**Discussion by alspdx** [DefinitelyTyped/DefinitelyTyped#55540] [Issues with a @types package]
Hi, I found an issue with the `@types/react-table` package. 

In the declaration for the `Accessor` function (used by `ColumnWithLooseAccessor`), the required `sub` parameter requires a property `data`. In the `useTable` hook code, a `row` is passed in as the third parameter, which looks to be a `Row<D>` type. 

```ts
/* declaration in `@types/react-table` */
export type Accessor<D extends object> = (
    originalRow: D,
    index: number,
    sub: {
        subRows: D[];
        depth: number;
        data: D[]; // does not exist on `Row<D>`
    },
) => CellValue;

/* how accessor is used in the `useTable` hook: */
// If the column has an accessor, use it to get a value
if (column.accessor) {
    row.values[column.id] = column.accessor(
        originalRow,
        rowIndex,
        row, // <-- type `Row<D>`
        parentRows,
        data
    )
}
```

The library itself isn't affected by this, but my team is writing a plugin for our own use to output data in the table to an Excel workbook, and would like to use the accessor function to get the same cell data as the table. Passing a `Row<D>` object as that function parameter results in a TS error due to the lack of `row.data` in the `Row<D>` type. The parameter is required by the type declaration — I'm not sure if it makes a difference in the types, but interesting to note is that [it is not mentioned in the column options documentation for accessor](https://react-table.tanstack.com/docs/api/useTable#column-options). 

Edit: I'd be happy to submit a PR to update the type declaration.

[a comment by an automated account is not shown]

**Comment by ggascoigne** [comment]
Just looking at this.  Looks like the `sub` got removed an age ago, and the new parameters got added in January.  Neither are part of the documented interface.  I'd be happy to take a PR for this.
